### PR TITLE
(hopefully) fix macos9 compiler

### PR DESCRIPTION
**Pull request by ChrisNonyminus**
This should probably fix the issues that were going on with getting it to work.
What I had to do was update the assembler syntax converter, as its detection of branches wasn't working right (the regex was expecting 2 tabs when there was only one).

### CODE REVIEW

File: frontend/next.config.js [hunk 1]
```diff
@@ -29,7 +29,7 @@ const removeImports = require("next-remove-imports")({
 })
 const nextTranslate = require("next-translate")
 
-const mediaUrl = new URL(process.env.MEDIA_URL ?? "http://localhost")
+const mediaUrl = new URL(process.env.MEDIA_URL != null ? process.env.MEDIA_URL : "http://localhost")
```
Comment (nanaian) [inline frontend/next.config.js:32]: Can you explain the reasoning for this change?

Comment (ChrisNonyminus) [reply frontend/next.config.js:32]: For some reason, the ``??`` won't mesh well with the latest javascript install on my machine.

Comment (nanaian) [reply frontend/next.config.js:32]: That's weird, what's your `node --version`?

Comment (ChrisNonyminus) [reply frontend/next.config.js:32]: v12.22.9

Comment (nanaian) [reply frontend/next.config.js:32]: Hm, `??` should be supported OK there. By "won't mesh well" what exactly do you mean? Did it complain of a syntax error?

Comment (ChrisNonyminus) [reply frontend/next.config.js:32]: Yes.

Comment (ChrisNonyminus) [reply frontend/next.config.js]: Updated node, removing the need for this workaround. Thus, I reverted the workaround.

**Review by nanaian — APPROVED** [review]

**Pull request merged**